lua/luasnip/extras/treesitter_postfix.lua

```diff
@@ -0,0 +1,317 @@
+local snip = require("luasnip.nodes.snippet").S
+local ts = require("luasnip.extras._treesitter")
+local node_util = require("luasnip.nodes.util")
+local extend_decorator = require("luasnip.util.extend_decorator")
+local tbl = require("luasnip.util.table")
+local util = require("luasnip.util.util")
+
+---@param opts string|string[]|LuaSnip.extra.MatchTSNodeOpts
+---@return LuaSnip.extra.MatchTSNodeFunc
+local function generate_match_tsnode_func(opts)
+	if type(opts) == "string" then
+		opts = { query = opts }
+	elseif vim.tbl_isarray(opts) then
+		opts = { query = { "luasnip" }, select = { captures = opts } }
+	end
+
+	---@param parser LuaSnip.extra.TSParser
+	---@param pos { [1]: number, [2]: number }?
+	---@return TSNode?
+	return function(parser, pos)
+		return parser:captures_at_pos(
+			opts, --[[@as LuaSnip.extra.MatchTSNodeOpts]]
+			pos
+		)
+	end
+end
+
+---@param reparse boolean|string|nil
+---@param real_resolver function
+---@return fun(snippet, line_to_cursor, matched_trigger, captures):table?
+local function wrap_with_reparse_context(reparse, real_resolver)
+	if reparse == "live" then
+		return function(snippet, line_to_cursor, matched_trigger, captures)
+			local bufnr = vim.api.nvim_win_get_buf(0)
+			local cursor = vim.api.nvim_win_get_cursor(0)
+			local enter, leave =
+				ts.wrap_enter_and_leave_func(bufnr, matched_trigger)
+
+			enter()
+			local parser, source = vim.treesitter.get_parser(bufnr), bufnr
+			if parser == nil or source == nil then
+				return nil
+			end
+			local ret = real_resolver(
+				snippet,
+				line_to_cursor,
+				matched_trigger,
+				captures,
+				parser,
+				source,
+				bufnr,
+				{ cursor[1] - 1, cursor[2] - #matched_trigger - 1 }
+			)
+			leave()
+			return ret
+		end
+	end
+	if reparse == "copy" then
+		return function(snippet, line_to_cursor, matched_trigger, captures)
+			local bufnr = vim.api.nvim_win_get_buf(0)
+			local cursor = vim.api.nvim_win_get_cursor(0)
+			local parser, source =
+				ts.reparse_buffer_after_removing_match(bufnr, matched_trigger)
+			if parser == nil or source == nil then
+				return nil
+			end
+			local ret = real_resolver(
+				snippet,
+				line_to_cursor,
+				matched_trigger,
+				captures,
+				parser,
+				source,
+				bufnr,
+				{ cursor[1] - 1, cursor[2] - #matched_trigger - 1 }
+			)
+			parser:destroy()
+			return ret
+		end
+	end
+
+	return function(snippet, line_to_cursor, matched_trigger, captures)
+		local bufnr = vim.api.nvim_win_get_buf(0)
+		local cursor = vim.api.nvim_win_get_cursor(0)
+		local parser, source = vim.treesitter.get_parser(bufnr), bufnr
+		if parser == nil or source == nil then
+			return nil
+		end
+		return real_resolver(
+			snippet,
+			line_to_cursor,
+			matched_trigger,
+			captures,
+			parser,
+			source,
+			bufnr,
+			{ cursor[1] - 1, cursor[2] - #matched_trigger - 1 }
+		)
+	end
+end
```

Comment:
Making `reparse` a whole object seems a bit overcomplicated (although I can definitely see the appeal), since it's just a wrapper around `nvim_buf_set_text` (either to `""` or to `matched_trigger`).
What do you think of replacing `enter`/`leave` with the direct calls?

Reply:
Yeah.

Reply:
TY!